src/GridFieldEditableColumns.php

```diff
@@ -130,12 +130,22 @@ public function handleSave(GridField $grid, DataObjectInterface $record)
         /** @var GridFieldOrderableRows $sortable */
         $sortable = $grid->getConfig()->getComponentByType(GridFieldOrderableRows::class);
 
+        // Fetch the items before processing them
+        $ids = array_keys($value[self::POST_KEY]);
+        $itemsCollection = ArrayList::create($list->filter('ID', $ids)->toArray());
+
         foreach ($value[self::POST_KEY] as $id => $fields) {
             if (!is_numeric($id) || !is_array($fields)) {
                 continue;
             }
 
-            $item = $list->byID($id);
+            // Find the item from the fetched collection of items
+            $item = $itemsCollection->filter('ID', $id)->first();
+
+            // Skip items that does not have any changed fields
+            if (!$this->isChanged($item, $fields)) {
+                continue;
+            }
```

Comment:
```suggestion
            if (!$item || !$this->isChanged($item, $fields) || !$item-canEdit()) {
                continue;
            }
```
Conbine this with the conditional block below